### PR TITLE
Update rtd config to v2

### PR DESCRIPTION
**Pull request by rashidnhm**
This PR updates the RTD configuration to use the new v2 syntax.

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+      path: .
+  system_packages: true
 
-# Don't build any extra formats
-formats:
-    - none
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.8"
```
Comment (albi3ro) [inline .readthedocs.yml:16]: Should we also bump the python version? I think we'll be dropping python 3.8 support in the next release.

File: .readthedocs.yml [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+sphinx:
+  configuration: doc/conf.py
 
 python:
-    version: 3.8
+  install:
+    - requirements: doc/requirements.txt
+    - method: pip
+      path: .
+  system_packages: true
```
Comment (timmysilv) [inline .readthedocs.yml:11]: is this necessary?

**Review by timmysilv — APPROVED** [review]
+1 to Christina's comment re: 3.9, otherwise lgtm!

File: .readthedocs.yml (hunk 2 above)
Comment (rashidnhm) [reply .readthedocs.yml]: Originally saw it being used in the lightning repo. But removed as it's being deprecated!

File: .readthedocs.yml (hunk 1 above)
Comment (rashidnhm) [reply .readthedocs.yml]: Bumped!

**Review by albi3ro — APPROVED** [review]
👍 thanks :)

**Pull request merged**